spring-integration-stomp/src/main/java/org/springframework/integration/stomp/outbound/StompMessageHandler.java

```diff
@@ -104,9 +106,11 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(final Message<?> message) throws Exception {
-		if (!this.isRunning()) {
+		if (!this.isRunning() || !this.stompSessionManager.isConnected()) {
+			this.running = false;
+			this.stompSession = null;
 			throw new MessageDeliveryException(message, "The StompMessageHandler [" + getComponentName() +
-					"] hasn't been connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
+					"] isn't connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
 		}
```

Comment:
Right; I understand; I am just a bit queasy about throwing an exception from an outbound adapter because the connection is not open yet.